File: Java/Android/OpenGLES/source/core/src/loon/utils/xml/XMLAttribute.java
package loon.utils.xml;


/**
 * Copyright 2008 - 2011
 * 
 * Licensed under the Apache License, Version 2.0 (the "License"); you may not
 * use this file except in compliance with the License. You may obtain a copy of
 * the License at
 * 
 * http://www.apache.org/licenses/LICENSE-2.0
 * 
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
 * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
 * License for the specific language governing permissions and limitations under
 * the License.
 * 
 * @project loon
 * @author cping
 * @email：javachenpeng@yahoo.com
 * @version 0.1
 */
public class XMLAttribute {

	private String name;

	private String value;

	XMLElement element;

	XMLAttribute(String n, String v) {
		this.name = n;
		this.value = v;
	}
	
	public XMLElement getElement() {
		return element;
	}

	public String getValue() {
		return this.value;
	}

	public int getIntValue() {
		try {
			return Integer.parseInt(this.value);
		} catch (Exception ex) {
			throw new RuntimeException("Attribute '" + this.name
					+ "' has value '" + this.value
					+ "' which is not an integer !");
		}
	}

	public float getFloatValue() {
		try {
			return Float.parseFloat(this.value);
		} catch (Exception ex) {
			throw new RuntimeException("Attribute '" + this.name
					+ "' has value '" + this.value
					+ "' which is not an float !");
		}
	}

	public double getDoubleValue() {
		try {
			return Double.parseDouble(this.value);
		} catch (Exception ex) {
			throw new RuntimeException("Attribute '" + this.name
					+ "' has value '" + this.value
					+ "' which is not an double !");
		}
	}

	public boolean getBoolValue() {
		if (value == null) {
			return false;
		}
		return "true".equalsIgnoreCase(value) || "yes".equalsIgnoreCase(value);
	}

	public String getName() {
		return this.name;
	}

}

File: C#/WindowsPhone/LGame-XNA-lib/Loon.Utils.Xml/XMLAttribute.cs
using System;
using Loon.Java;
namespace Loon.Utils.Xml {
	
	public class XMLAttribute {
	
		private string name;
	
		private string value_ren;
	
		internal XMLElement element;
	
		internal XMLAttribute(string n, string v) {
			this.name = n;
			this.value_ren = v;
		}
	
		public XMLElement GetElement() {
			return element;
		}
	
		public string GetValue() {
			return this.value_ren;
		}
	
		public int GetIntValue() {
			try {
				return Int32.Parse(this.value_ren);
			} catch (Exception) {
				throw new Exception("Attribute '" + this.name
						+ "' has value '" + this.value_ren
						+ "' which is not an integer !");
			}
		}
	
		public float GetFloatValue() {
			try {
				return Single.Parse(this.value_ren,JavaRuntime.NumberFormat);
			} catch (Exception) {
				throw new Exception("Attribute '" + this.name
						+ "' has value '" + this.value_ren
						+ "' which is not an float !");
			}
		}
	
		public double GetDoubleValue() {
			try {
				return ((Double )Double.Parse(this.value_ren,JavaRuntime.NumberFormat));
			} catch (Exception) {
				throw new Exception("Attribute '" + this.name
						+ "' has value '" + this.value_ren
						+ "' which is not an double !");
			}
		}
	
		public bool GetBoolValue() {
			if (value_ren == null) {
				return false;
			}
			return "true".Equals(value_ren,StringComparison.InvariantCultureIgnoreCase) || "yes".Equals(value_ren,StringComparison.InvariantCultureIgnoreCase);
		}
	
		public string GetName() {
			return this.name;
		}
	
	}
}
